### PR TITLE
Add docs for upcoming Permission Boundary feature

### PR DESCRIPTION
**Pull request by renebrandel**
_Issue #, if available:_

_Description of changes:_
Add docs for the Permission Boundary feature

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by medelman17 — APPROVED** [review]
LGTM

File: docs/cli/teams/commands.md [hunk 1]
```diff
@@ -22,11 +22,12 @@ amplify env
 
 | Command  | Description  |
 |---|---|
-| [`amplify env add`](#add-a-new-environment) | Adds a new environment |
+| [`amplify env add [--permissions-boundary]`](#add-a-new-environment) | Adds a new environment |
```
Comment (edwardfoyle) [inline docs/cli/teams/commands.md:25]: should specify that the parameter takes an argument and isn't just a flag:
```suggestion
| [`amplify env add [--permissions-boundary <IAM policy ARN>]`](#add-a-new-environment) | Adds a new environment |
```

File: docs/cli/teams/commands.md [hunk 2]
```diff
@@ -22,11 +22,12 @@ amplify env
...
 | [`amplify env pull [--restore]`](#pull-the-environment-from-the-cloud) | Pulls the current environment from the cloud |
 | [`amplify env checkout <env-name> [--restore]`](#checkout-an-environment) | Switches to the selected environment |
 | [`amplify env list [--details] [--json]`](#list-environments) | Displays a list of all the environments |
 | [`amplify env get --name <env-name>`](#show-environment-details) | Displays the environment details |
+| [`amplify env update [--permissions-boundary]`](~/cli/usage/permissions-boundary.md) | Updates the environment's IAM Permissions Boundary |
```
Comment (edwardfoyle) [inline docs/cli/teams/commands.md:30]: same here:
```suggestion
| [`amplify env update [--permissions-boundary <IAM policy ARN>]`](~/cli/usage/permissions-boundary.md) | Updates the environment's IAM Permissions Boundary |
```

File: docs/cli/usage/permissions-boundary.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+The IAM Permissions Boundary will apply to all IAM Roles created by Amplify. This includes the "auth role" assumed by users that log into the app and the "unauth role" assumed by guest users. It also applies to Lambda execution roles, Cognito user group roles, and any role configured in a [custom resource stack](~/cli/usage/customcf.md).
+
+Amplify CLI does set up a new IAM Policy to use as a Permissions Boundary. A Permissions Boundary is an IAM Policy and can be created following the guide [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create-console.html). This IAM Policy, to be used as a Permissions Boundary, must be configured outside of Amplify CLI. This is usually part of an AWS Organization rule or other corporate governance requirement. Once you have created an IAM Policy to use as a Permissions Boundary, copy the IAM Policy ARN for the next steps.
```
Comment (edwardfoyle) [inline docs/cli/usage/permissions-boundary.md:10]: First sentence should say `Amplify CLI does not set up...`. Although I think the first sentence can be removed as the 3rd sentence basically says the same thing

**Pull request merged**